### ISSUE TITLE
`torch.compile` produces wrong results when `F.layer_norm` is fused with `reciprocal`/`rsqrt` epilogue

### ISSUE THREAD

**Issue by wuyii8941** [pytorch/pytorch#186577]
## Summary

When Inductor fuses `F.layer_norm` with a subsequent division-like pointwise op (`reciprocal`, `rsqrt`, `1/sqrt`), the generated kernel produces significantly wrong results. Each op alone compiles correctly; only the fused combination triggers the bug. `aot_eager` produces correct results, confirming this is an Inductor codegen issue.

## Minimal reproducer

```python
import torch
import torch.nn.functional as F

x = torch.randn(64, 128, device='cuda')

def fn(x):
    return torch.reciprocal(torch.abs(F.layer_norm(x, [128])) + 1e-6)

eager = fn(x)

torch._dynamo.reset()
aot = torch.compile(fn, backend='aot_eager')(x)

torch._dynamo.reset()
compiled = torch.compile(fn, backend='inductor')(x)

print(f"eager vs aot_eager: {(eager - aot).abs().max().item():.4e}")    # 0.0000e+00
print(f"eager vs inductor:  {(eager - compiled).abs().max().item():.4e}")  # ~9e-01
```

## Affected ops (epilogue after `layer_norm`)

| Epilogue op | Triggers? | Max diff |
|-------------|-----------|----------|
| `reciprocal` | Yes | ~80 (shape dependent) |
| `rsqrt` | Yes | ~877 (with cummax prefix) |
| `1/sqrt` (manual) | Yes | ~0.05 |
| `exp` | No | ~7e-6 |
| `tanh` | No | ok |

## Trigger conditions

- **Op pattern**: `F.layer_norm(x, [N])` followed by `reciprocal`, `rsqrt`, or `1/sqrt`
- **Shape dependent**: larger tensors trigger more reliably; small tensors (e.g., `(2, 256)`) may not trigger
- **dtype**: primarily float32; float16 also affected; float64 and bfloat16 appear unaffected
- **Backend**: Inductor only; `aot_eager` is correct
- **Each op alone is correct**: `layer_norm` alone is correct (diff ~7e-7), `reciprocal`/`rsqrt` alone on LN output is correct (diff = 0)
- **Manual normalization does not trigger**: `x / sqrt(mean(x^2) + eps)` + `rsqrt` is fine — only `F.layer_norm` triggers

## Amplified by upstream scan ops

When `cummax`/`cummin`/`cumsum` precedes the pattern, the error is amplified dramatically because the scan output distribution differs from random normal:

```python
x = torch.randn(68, 4, 14, 9, device='cuda')

def fn(x):
    t = torch.cummax(x, dim=-1)[0]         # scan op
    t = F.layer_norm(t, [9])                # layer_norm
    return torch.rsqrt(torch.abs(t) + 1e-6) # rsqrt epilogue

eager = fn(x)
torch._dynamo.reset()
compiled = torch.compile(fn, backend='inductor')(x)
print((eager - compiled).abs().max().item())  # ~877
```

## Shape sensitivity

```
# reciprocal(|LN(x)| + eps), varying shape:
(32, 64)      → diff=0.011
(64, 128)     → diff=0.82
(32, 32, 32)  → diff=81
(8, 4, 14, 9) → diff=0.077
```

## Environment

- PyTorch: 2.13.0.dev20260601+cu126
- GPU: NVIDIA Tesla T4 (CUDA 12.6)
- OS: Linux

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by eellison** [comment]
As per the fuzzer guidelines - please include in your repro a comparison to the higher precision baseline:

```
import torch
import torch.nn.functional as F

x = torch.randn(64, 128, device='cuda')

def fn(x):
    return torch.reciprocal(torch.abs(F.layer_norm(x, [128])) + 1e-6)

# fp64 reference (ground truth)
ref = fn(x.double()).float()

eager = fn(x)

torch._dynamo.reset()
compiled = torch.compile(fn, backend='inductor')(x)

eager_max = (eager - ref).abs().max().item()
compiled_max = (compiled - ref).abs().max().item()

print(f"eager    vs fp64  max: {eager_max:.4e}")
print(f"inductor vs fp64  max: {compiled_max:.4e}")
print(f"inductor closer (max): {compiled_max < eager_max}")
```

eager    vs fp64  max: 1.6357e-02
inductor vs fp64  max: 5.6152e-03
inductor closer (max): True

https://github.com/pytorch/pytorch/issues/new?template=pt2-bug-report.yml

<img width="988" height="169" alt="Image" src="https://github.com/user-attachments/assets/fefaf1a4-19a2-463c-9f5e-a8ba0e0a7167" />

cc @jansel , can you update your agent with instructions about this, especially for fuzzed issues?

**Comment by jansel** [comment]
@eellison done
